From: Chad Monroe <chad@monroe.io>
Date: Thu, 27 Apr 2023 16:56:32 -0700
Subject: [PATCH] net: phy: mxl: disable SGMII auto-negotiation

Always disable SGMII auto-negotiation as it will be disabled on
the MAC side. Checking that this persists from the update interface
poller is important as some revisions of GPY211 (at least GPY211C0)
seem to randomly reset this value back to default during operation.

Signed-off-by: Chad Monroe <chad@monroe.io>
---
 drivers/net/phy/mxl-gpy.c |   10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

--- a/drivers/net/phy/mxl-gpy.c
+++ b/drivers/net/phy/mxl-gpy.c
@@ -370,17 +370,11 @@ static void gpy_update_interface(struct
 	case SPEED_100:
 	case SPEED_10:
 		phydev->interface = PHY_INTERFACE_MODE_SGMII;
-		if (gpy_sgmii_aneg_en(phydev))
-			break;
-		/* Enable and restart SGMII ANEG for 10/100/1000Mbps link speed
-		 * if ANEG is disabled (in 2500-BaseX mode).
-		 */
 		ret = phy_modify_mmd(phydev, MDIO_MMD_VEND1, VSPEC1_SGMII_CTRL,
-				     VSPEC1_SGMII_ANEN_ANRS,
-				     VSPEC1_SGMII_ANEN_ANRS);
+				     VSPEC1_SGMII_CTRL_ANEN, 0);
 		if (ret < 0)
 			phydev_err(phydev,
-				   "Error: Enable of SGMII ANEG failed: %d\n",
+				   "Error: Disable of SGMII ANEG failed: %d\n",
 				   ret);
 		break;
 	}
